cli/azd/pkg/auth/manager.go

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package auth
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/public"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// TODO(azure/azure-dev#710): Right now, we re-use the App Id of the `az` CLI, until we have our own.
+//
+// nolint:lll
+// https://github.com/Azure/azure-cli/blob/azure-cli-2.41.0/src/azure-cli-core/azure/cli/core/auth/identity.py#L23
+const cAZD_CLIENT_ID = "04b07795-8ddb-461a-bbee-02f9e1bf7b46"
+
+// cCurrentUserKey is the key we use in config for the storing identity information of the currently logged in user.
+const cCurrentUserKey = "auth.account.currentUser"
```

Comment:
@wbreza @ellismg I think this currently creates
```json
{
"auth": {
 "account": {
    "currentUser": {
       "homeAccountId":"",
       "tenantId":"",
       "clientId":""
     }
  }
}
}
```

This is perfectly fine for most applications, but I wonder if we want a more user-friendly format for CLI users to be able to edit directly. I would lean towards having a single nesting like VSCode:

```json
"auth.account.currentUser": {
       "homeAccountId":"",
       "tenantId":"",
       "clientId":""
}
```

Basically, only one-level of object nesting is allowed. `Set(key string, value [map or primitive])`

Reply:
Your assumption on how the file is saved here is correct.  Happy to go make some changes here.  I do worry a little about not being able to have the value be an object which itself contains other pieces of data, but maybe it won't be a problem in practice.

Do you want to figure this out as part of this change or do you want me to defer that?

Reply:
Just wanted your thoughts @ellismg as the first-consumer outside of `azd.defaults`. Created #1091 to get consensus and decide if we want to implement this.